pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -124,21 +130,27 @@
                 nativeQueries = new ArrayList<NativeQuery>();
               }
 
-              nativeQueries.add(new NativeQuery(nativeSql.toString(), toIntArray(bindPositions),isCurrentReWriteCompatible));
+              nativeQueries.add(new NativeQuery(nativeSql.toString(),
+                  toIntArray(bindPositions), DMLCommand.createStatementTypeInfo(
+                  current, isBatchedReWriteConfigured, isCurrentReWriteCompatible,
+                  isReturningPresent, isAutoCommit, nativeQueries.size())));
             }
             // Prepare for next query
             if (bindPositions != null) {
               bindPositions.clear();
             }
             nativeSql.setLength(0);
+            current = DMLCommandType.BLANK;
+            isReturningPresent = false;
           }
           break;
 
         case 'i':
           if (Parser.parseInsertKeyword(aChars, i)) {
```

Comment:
yes good point. I'll re-factor the switch to convert the char to upper case. Which will remove the duplicate lower case block.